src/domain/core/src/services/push_status_service.rs

```diff
@@ -0,0 +1,81 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::collections::HashMap;
+
+use internal_error::InternalError;
+use opendatafabric::{DatasetHandle, DatasetRefRemote};
+
+use crate::utils::metadata_chain_comparator::CompareChainsResult;
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[async_trait::async_trait]
+pub trait PushStatusService: Send + Sync {
```

Comment:
Why do we need another service? It seems to me that this functionality would fit just fine in the existing `PushService`. They both operate on the same logical level of push aliases.